app/utils/injectSaga.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './sagaInjectors';
+
+/**
+ * Dynamically injects a saga, passes component's props as saga arguments
+ *
+ * @param {string} name Name of the saga
+ * @param {function} saga A root saga that will be injected
+ * @param {string} [mode] By default ('restart-on-remount') the saga will be started on component mount and cancelled with
```

Comment:
since mode is used in many files, maybe we should create constant ? ie 
```js
// utils/constants.js
export const RESTART_ON_REMOUNT = '@@saga/restart-on-remount'
export const DAEMON = '@@saga/daemon'
// ...

// ...component...
import {
  RESTART_ON_REMOUNT
} from 'utils/constants'
```